### ISSUE TITLE
version mismatch with mongo-ruby-driver 2.0.x

### ISSUE THREAD

**Issue by m-mizutani** [fluent/fluent-plugin-mongo#47]
It seems that API call of mongo-ruby-driver is not matched with latest version (2.0.1)

For example, creating client code in out_mongo.rb is below.

```ruby
db = Mongo::MongoClient.new(@host, @port, @connection_options).db(@database)
```

However example to create client in [the document](http://docs.mongodb.org/ecosystem/tutorial/ruby-driver-tutorial/) is below.

```ruby
client = Mongo::Client.new([ '127.0.0.1:27017' ], :database => 'music')
```

Then, fluent-plugin-mongo has following error with mongo-ruby-driver 2.0.1.

```
2015-03-30 15:22:40 +0000 [error]: unexpected error error_class=NameError error=#<NameError: uninitialized constant Mongo::MongoClient>
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluent-plugin-mongo-0.7.8/lib/fluent/plugin/out_mongo.rb:215:in `get_connection'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluent-plugin-mongo-0.7.8/lib/fluent/plugin/out_mongo.rb:192:in `get_or_create_collection'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluent-plugin-mongo-0.7.8/lib/fluent/plugin/out_mongo.rb:87:in `start'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/agent.rb:67:in `block in start'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/agent.rb:66:in `each'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/agent.rb:66:in `start'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/root_agent.rb:104:in `start'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/engine.rb:201:in `start'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/engine.rb:151:in `run'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/supervisor.rb:481:in `run_engine'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/supervisor.rb:140:in `block in start'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/supervisor.rb:266:in `call'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/supervisor.rb:266:in `main_process'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/supervisor.rb:241:in `block in supervise'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/supervisor.rb:240:in `fork'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/supervisor.rb:240:in `supervise'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/supervisor.rb:134:in `start'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/lib/fluent/command/fluentd.rb:167:in `<top (required)>'
  2015-03-30 15:22:40 +0000 [error]: /usr/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
  2015-03-30 15:22:40 +0000 [error]: /usr/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
  2015-03-30 15:22:40 +0000 [error]: /var/lib/gems/2.2.0/gems/fluentd-0.12.7/bin/fluentd:6:in `<top (required)>'
  2015-03-30 15:22:40 +0000 [error]: /usr/local/bin/fluentd:23:in `load'
  2015-03-30 15:22:40 +0000 [error]: /usr/local/bin/fluentd:23:in `<main>'
```

I'm not an expert in them and I'm sorry if I miss something.

Thanks.

**Comment by repeatedly** [comment]
I tried to support mongo 2.x series but I hit weird situation.
I want to disable SSL access but ruby driver try to connect to mongodb using SSL.
Run following code:

```client = Mongo::Client.new(['127.0.0.1:27017'], :ssl => false)```

got following logs:

```
D, [2015-03-31T18:33:58.664761 #15021] DEBUG -- : MONGODB | Adding 127.0.0.1:27017 to the cluster. | runtime: 0.0029ms
D, [2015-03-31T18:33:58.665829 #15021] DEBUG -- : MONGODB | QUERY | namespace=admin.$cmd selector={:ismaster=>1} flags=[] limit=-1 skip=0 project=nil | runtime: 0.9251ms
D, [2015-03-31T18:33:58.665867 #15021] DEBUG -- : MONGODB | SSL_connect SYSCALL returned=5 errno=0 state=SSLv2/v3 read server hello A | runtime: 0.0010ms
```

I want to know why this happens...

Using ruby driver 1.x is better for now.

**Comment by eiszfuchs** [comment]
I've the same problem now. I just wanted to try `fluentd`. Shouldn't the `gemspec` tell `gem` not to use v2.x then?
https://github.com/fluent/fluent-plugin-mongo/blob/master/fluent-plugin-mongo.gemspec#L20

Is there a quicker way to fix this issue than to wait until a new release?

**Comment by eiszfuchs** [comment]
To aswer my own question for those who run into the same issue:
Oh, and check your dependencies first before downgrading.

    $ gem uninstall mongo

    You have requested to uninstall the gem:
            mongo-2.0.1

    fluent-plugin-mongo-0.7.8 depends on mongo (>= 1.9.0)
    If you remove this gem, these dependencies will not be met.
    Continue with Uninstall? [yN]  y
    Successfully uninstalled mongo-2.0.1
    $ gem install mongo -v 1.9.0

**Comment by repeatedly** [comment]
I tried to update mongo gem to v2 but failed, because v2 lacks some v1 APIs.
I decided to use only v1 for gem dependency.

**Comment by repeatedly** [comment]
I gave up mongo gem v2 support.
I just released v0.7.9 which forces to use mongo gem v1.

**Comment by m-mizutani** [comment]
It works for me. Thanks!
